dotnet/src/Svg.cs

```diff
@@ -0,0 +1,162 @@
+            Svg.GetImage(fileName, isSpot: true, bypassSizeCheck: true);
+    }
+
+    /// <summary>
+    /// A hash that changes when any of the icons change or one gets added, removed, etc. Calculated in GetImage() by replacing this
+    /// value with a hash of the old value combined with the file contents. Thus relies on the fact that C# guarantees in-order execution
+    /// of static initializers (because the files have to be in the same order every time). The JS uses this value as a cache breaker when
+    /// downloading any SVG icon.
+    /// </summary>
+    public static string CombinedCacheBreaker { get; private set; } = string.Empty;
+
+    /// <summary>
+    /// Gets an <see cref="SvgImage"/> for caching and reuse.
+    /// </summary>
+    /// <param name="fileName">The filename to grab, defaults to the caller's name via <see cref="CallerMemberNameAttribute"/>.</param>
+    /// <param name="isSpot">Whether the image is a Spot or not.</param>
+    /// <param name="bypassSizeCheck">Whether to allow bypassing the size check, for things we know not to repeat in a page (stil shouldn't be huge).</param>
+    /// <returns>The <see cref="SvgImage"/> to cache.</returns>
+    internal static SvgImage GetImage([CallerMemberName] string? fileName = null, bool isSpot = false, bool bypassSizeCheck = false)
+    {
+        if (string.IsNullOrEmpty(fileName))
+        {
+            return Empty;
+        }
+
+        try
+        {
+            var imageString = Helpers.GetSvg(fileName!, isSpot);
+#if DEBUG
+            // Okay so now we have the svg asset loaded into the string,
+            // if we're on DEBUG let's do some gut checking on the .svg itself.
+            // This way we throw in dev.
+            if (imageString is null)
+            {
+                throw new Exception($"Unable to find {fileName} in embedded resources.");
+            }
+
+            if (imageString.Length > MaxReasonableSize && !bypassSizeCheck)
+            {
+                throw new Exception($"{fileName} is larger than the maximum allowed SVG icon size: {MaxReasonableSize} bytes.");
+            }
+
+            if (imageString.Contains("<!--"))
+            {
+                throw new Exception($"{fileName} contains <!-- --> style comments. Please ensure this file is minified properly.");
+            }
+#endif
+            CombinedCacheBreaker = Helpers.ToSha256Hash(CombinedCacheBreaker + imageString).Substring(0, 12);
+            return new SvgImage(imageString);
+        }
+        catch (Exception ex)
+        {
+#if DEBUG
+            // If the file wasn't able to be loaded at all, we should just throw.
+            // This throw should never happen on prod since people should be checking after they add a new SVG
+            // and the entry for it in this file, but you know, just in case.
+            throw new Exception($"Unable to open {fileName}.svg (isSpot: {isSpot})", ex);
+#else
+            return Empty;
```

Comment:
Yeah, the original code (from Core) logged an exception here using a globally static logger instance. I think I have two options here:

1. introduce a way to pass in a logger (painful, since these are static methods instantiated at startup)
2. leave it as-is since exceptions will be (extremely likely) caught during dev

Thoughts? I'm not sure of any good workarounds for this.

Reply:
Understood. I am happy to leave it like that for now - perhaps add a TODO comment though mentioning that we should either log or avoid swallowing this?

Side note (rambling - feel free to ignore):
I think there might be a 3rd option which is removing the try/catch statement all together from here.
I find the code a bit defensive in this context since the consumers of the Svg class can only call methods for which we know we have a valid SVG (typos, etc.. will be caught at compile time). 
The only reason I can think an exception could happen here is because of some hiccups in the reading/loading process (if we want to handle those (probably very rare) cases we could push the exception control flow down a level in the Helpers GetSvg method and return null in case the reader blows up. I guess where I am getting at is that this try/catch statement mixes up 2 concerns (loading file, validate file).